Feature | change request | main service other input field

omarCreativeDev opened the pull request:
- Add the GDS conditionally revealing content for the `Other` text input on each `Other` radio element
- Ensure `maxlength` validator is set on the other input text field only if something is entered as its an optional field.
- Update the registration `POST` model request, associated interfaces etc
- Display new entered `Other` field on the confirm your workplace details summary page

ChazUK commented on src/app/features/registration/select-main-service/select-main-service.component.html:
```diff
@@ -27,20 +29,53 @@ <h2 class="govuk-fieldset__heading">
+                  </label>
+                </div>
+
+                <!-- conditional content -->
```
not sure if we need the comment

omarCreativeDev replied:
done

ChazUK commented on src/app/features/registration/select-main-service/select-main-service.component.html:
```diff
@@ -27,20 +29,53 @@ <h2 class="govuk-fieldset__heading">
+
+                <!-- conditional content -->
+                <div
+                  *ngIf="service.other && form.get('workplaceService').value === service.id.toString()"
```
just checked the GOV.UK Framework and we should actually be putting the class `govuk-radios__conditional--hidden` on hidden conditionals, same with checkboxes govuk-checkboxes__conditional--hidden

omarCreativeDev replied:
cheers for spotting, I am now setting `govuk-radios__conditional--hidden` css class on markup to only show other input field if associated radio is selected, otherwise add all other conditional revealing content to the dom which will help with accessibility.

I wiil create a bug and handle this in all other areas seperately

ChazUK commented on src/app/features/registration/select-main-service/select-main-service.component.ts:
```diff
@@ -78,32 +90,49 @@ export class SelectMainServiceComponent implements OnInit, OnDestroy {
     ];
   }
 
+  private subscribeToUserSelection(): void {
```
I don't think we want to reset the users input when they switch radio's. There was a discussion somewhere on slack but it's disappeared into the ether.

omarCreativeDev replied:
good spot! as discussed i am now looping over the workplaces with the `other` flag and if present im creating x number of text fields in the form

omarCreativeDev commented:
> mainServiceOther

cheers @wozzer72

omarCreativeDev commented:
hey @ChazUK I have addressed all comments you made in my recent commit/s. I will address the conditionally revealing class stuff globally on a seperate ticket. Let me know if all ok.

omarCreativeDev commented:
![](https://github.trello.services/images/mini-trello-icon.png) [(8) Change Request - Main Service "Other" Input field](https://trello.com/c/GkKQK1WE/78-8-change-request-main-service-other-input-field)

ChazUK commented on src/app/features/registration/confirm-workplace-details/confirm-workplace-details.component.html:
```diff
@@ -54,6 +54,20 @@ <h2 class="govuk-heading-m govuk-!-margin-bottom-7">Are these details correct?</
           </a>
         </dd>
       </div>
+      <div class="govuk-summary-list__row" *ngIf="workplaceService.otherName">
```
I would check with Rich but normally the Summary / Check Answers Page should display this in the same row as Main Service, instead of two rows as it's the same question

omarCreativeDev replied:
ok will check with rich and will tweak if needed in a new pr
